sklearn/cluster/_hdbscan/_linkage.pyx

```diff
@@ -150,6 +153,9 @@ cpdef cnp.ndarray[MST_edge_t, ndim=1, mode='c'] mst_from_data_matrix(
                 num_features
             )
 
+            if alpha != 1.0:
+                pair_distance /= alpha
```

Comment:
Why can't we just do this?
```suggestion
            pair_distance /= alpha
```

Just an optimization because division are costly and of no use in this case?

Reply:
That paired with the fact that the conditional was present in the original code and I didn't bother too much to figure out whether it was worth changing or not 😅 . Overall I prefer your suggestion due to code simplicity (no need to be greedy for at most a few instruction cycles).